libs/core/src/components/button/button.ts

```diff
@@ -143,9 +143,14 @@ export class GdsButton<ValueT = any> extends GdsFormControlElement<ValueT> {
 
     const tag = this.#isLink ? literal`a` : literal`button`
 
+    const classList = Object.entries(buttonClasses)
```

Comment:
The reason for the change here was that all classes would add a empty space in the start and end of the `class string` 

For example ` button primary positive ` instead of `button primary positive` do you think we should revert this or use any other method perhaps ?

Reply:
Yeah, I think we should stick with the Lit directive. Lit adds some memoization to help with performance (ie, cache the result to avoid re-computing the class list on every render), and the extra spaces shouldn't cause any issues.